src/main/java/duke/command/FindCommand.java

```diff
@@ -0,0 +1,47 @@
+package duke.command;
+
+import duke.Storage;
+import duke.TaskList;
+import duke.Ui;
+import duke.task.Task;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class FindCommand extends Command {
+
+    private final String findString;
+
+    public FindCommand(String findString) {
+        this.findString = findString;
+    }
+
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
+        if (tasks.isEmpty()) {
+            ui.printWithDivider("Your list is empty!");
+        } else {
+            List<Task> filteredTask = tasks.getTasks().stream()
+                    .filter(task -> task.getTaskName().contains(findString))
+                    .collect(Collectors.toList());
```

Comment:
It might be better to abstract this method within `TaskList`, which might also work better with the code below.

Reply:
Similar to `ListCommand`, I think this is really useful 👍